core/src/clients/snyk/snyk_client.rs

```diff
@@ -0,0 +1,223 @@
+
+        println!("{}", display_msg);
+        let response: Result<Option<OrgProjects>, HyperError> = post(
+            url.as_str(),
+            ContentType::Json,
+            snyk_token.as_str(),
+            None::<String>,
+        ).await;
+
+        match response {
+            Ok(option) => return Ok(option),
+            Err(err) => return Err(SnykProviderError::SnykConnection(format!("Error with: {}, \nadditional errors: {}", display_msg, err))),
+        }
+    }
+
+    // Returns a tuple that contains a list of OrgProject objects and a list of any errors encountered
+    async fn get_projects_from_org_list(&self, snyk_token: String, orgs: Vec<Org>) -> (Vec<OrgProjects>, Vec<SnykProviderError>) {
+
+        let mut all_projects: Vec<OrgProjects> = Vec::new();
+        let mut errors: Vec<SnykProviderError> = Vec::new();
+        for org in orgs.iter() {
+            let org_id = org.id.clone().unwrap_or_else(|| "Missing".to_string());
+            let org_name = org.name.clone().unwrap_or_else(|| "Missing".to_string());
+            let results = self.get_projects_from_org(snyk_token.clone(), org_id.clone(), org_name.clone()).await;
+            
+            match results {
+                Ok(option_proj_json) => {
+                    match option_proj_json {
+                        Some(org_proj) => all_projects.push(org_proj),
+                        None => {
+                            let err_msg = SnykProviderError::SnykNoProjectsFoundError(format!("----------------------\nNo projects found for \n--org_id: {}, \n--org_name: {}", org_id, org_name));
+                            errors.push(err_msg);
+                        }
+                    }
+                },
+                Err(err) => errors.push(err),
+            }
+        }
+        return (all_projects, errors);
+    }
+
+
+    // Returns an sbom from Snyk based 
+    async fn get_sbom_from_snyk(&self, snyk_token: String, org_id: String, proj_id: String) -> Result<Option<HashMap<String, Value>>, SnykProviderError> {
+        let sbom_url = format!("{}/orgs/{}/projects/{}/sbom?version={}&format={}", API_V3_URL, org_id, proj_id, API_V3_VERSION, SBOM_FORMAT);
+        let display_msg = format!("----------------------\nFetching sbom from Snyk \nproject: {}, \nOrg: {}, \nUrl: {}", proj_id, org_id, sbom_url);
+        println!("{}", display_msg);
+        let response: Result<Option<HashMap<String, Value>>, HyperError> = get(
+            &sbom_url,
+            ContentType::Json,
+            snyk_token.as_str(),
+            None::<String>,
+        ).await;
+
+        match response {
+            Ok(sbom) => return Ok(sbom),
+            Err(err) => Err(SnykProviderError::SnykSBOMError(format!("Error with: {}, \nadditional errors: {}", display_msg, err)))
+        }
+    }
+
+
+}
+
+#[tokio::test]
+pub async fn test_get_orgs() {
```

Comment:
Does this get executed when you run `cargo test`.

Reply:
Yes they do, although they are apparently failing at the moment.. which is new from yesterday